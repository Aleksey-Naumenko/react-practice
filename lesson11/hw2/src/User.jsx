/* eslint-disable camelcase */
import React from 'react';
import PropTypes from 'prop-types';

const User = ({ data }) => {
    const { name, location, avatar_url } = data;

    return (
        <div className="user">
            <img alt="User Avatar" src={avatar_url} className="user__avatar" />
            <div className="user__info">
                <span className="user__name">{name}</span>
                <span className="user__location">{location}</span>
            </div>
        </div>
    );
};

User.propTypes = {
    data: PropTypes.shape({
        name: PropTypes.string.isRequired,
        location: PropTypes.string.isRequired,
        avatar_url: PropTypes.string.isRequired,
    }).isRequired,
};

export default User;