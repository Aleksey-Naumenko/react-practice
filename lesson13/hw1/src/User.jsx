import React, { Component } from 'react';

class User extends Component {
    state = {
        userName: '',
        userLocation: '',
        userAvatar: '',
        counter: 0
    }
    componentDidUpdate(prevProps, prevState) {
        console.log('UPDATE happened');
        console.log(prevProps);
        console.log(prevState);

        //this.fetchUserData(); // updates when render
        if (prevProps.match.url !== this.props.match.url) {
            this.fetchUserData(this.props.match.url);
        }
    }

    //   userPath = this.props.match.url

    componentDidMount() {
        this.fetchUserData(this.props.match.url);
    }

    fetchUserData = (userPath) => {
        // console.log(this.props.match.url);
        fetch(`https://api.github.com${userPath}`)
            .then(response => response.json())
            .then(user => this.setState({
                userName: user.name,
                userLocation: user.location,
                userAvatar: user.avatar_url
            }));
    }

    render() {
        const { userName, userLocation, userAvatar } = this.state;
        return (
            <div className="user">
                <img alt="User Avatar" src={userAvatar} className="user__avatar" />
                <div className="user__info">
                    <span className="user__name">{userName}</span>
                    <span className="user__location">{userLocation}</span>
                </div>
                <button onClick={() => this.setState({
                    counter: this.state.counter + 1
                })}>
                    Counter now is on {this.state.counter}
                </button>
            </div>
        );
    }
};

export default User;