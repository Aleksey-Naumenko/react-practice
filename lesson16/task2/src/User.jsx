import React, { Component } from 'react';
import { getUserData } from './usersGateway';

class User extends Component {
    state = {
        userData: null,
    };

    componentDidMount() {
        getUserData().then(userData =>
            this.setState({
                userData,
            }),
        );
    }

    render() {
        const { userData } = this.state;
        if (!userData) {
            return null;
        }
        const { name, location, avatar_url } = userData;
        return (
            <div className="user">
                <img
                    alt="User Avatar"
                    src={avatar_url}
                    className="user__avatar"
                />
                <div className="user__info">
                    <span className="user__name">{name}</span>
                    <span className="user__location">{location}</span>
                </div>
            </div>
        );
    }
}

export default User;