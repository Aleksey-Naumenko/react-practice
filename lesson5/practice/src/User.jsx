import React, { Component } from "react";
import PropTypes from "prop-types";

class User extends Component {
  state = {
    userData: null,
  };

  componentDidMount() {
    const { userId } = this.props;
    this.fetchUser(userId);
  };

  componentDidUpdate(prevProps) {
    console.log("==> User update");
    const { userId } = this.props;
    if (prevProps.userId !== userId) {
      this.fetchUser(userId);
    }
  };

  fetchUser(userId) {
    fetch(`https://api.github.com/users/${userId}`)
      .then((response) => response.json())
      .then((data) => {
        this.setState({
          userData: data,
        });
      });
  };

  render() {
    console.log("==> User render");
    const { userData } = this.state;
    if (!userData) {
      return null;
    }
    const { avatar_url, name, location } = userData;
    return (
      <div className="user">
        <img alt="User Avatar" src={avatar_url} className="user__avatar" />
        <div className="user__info">
          <span className="user__name">{name}</span>
          <span className="user__location">{location}</span>
        </div>
      </div>
    );
  }
};

User.propTypes = {
  userId: PropTypes.string.isRequired,
};

export default User;
