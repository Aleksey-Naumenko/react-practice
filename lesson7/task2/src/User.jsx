import React from 'react';
import './index.scss';


const User = ({ name, age }) => {
    return (
        <li className="user">
            <span className="user__name">{name}</span>
            <span className="user__age">{age}</span>
        </li>
    );
};

export default User;