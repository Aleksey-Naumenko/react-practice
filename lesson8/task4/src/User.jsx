import React, { Component } from 'react';


class User extends Component {
    constructor(props) {
        super(props);
        this.state = {
            user: null,
        }
    }

    componentDidMount() {
        this.fetchUser(this.props.userId);
    }

    fetchUser = userId => {
        fetch(`https://api.github.com/users/${userId}`)
        .then(res => res.json())
        .then(data => this.setState({
            user: data,
        }));
    }

    render() {
        const { user } = this.state;
        if (!this.state.user) return null;

        const { name, location, avatar_url} = user;

        return (
            <div className="user">
                <img
                    alt="User Avatar"
                    src={avatar_url}
                    className="user__avatar"
                />
                <div className="user__info">
        <span className="user__name">{name}</span>
                    <span className="user__location">{location}</span>
                </div>
            </div>
                )
            }
        }
        
export default User;