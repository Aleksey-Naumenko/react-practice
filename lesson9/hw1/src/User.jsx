import React from 'react';

const User = props => {
    return (
        <li className="user">
            <span className="user__name">{props.name}</span>
            <span className="user__age">{props.age}</span>
        </li>
    );
};

export default User;