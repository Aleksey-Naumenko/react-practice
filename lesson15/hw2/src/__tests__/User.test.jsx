import React from "react";
import User from '../User';
import { shallow } from 'enzyme';

describe('<User />', () => {
    it ('should render passed name', () => {
        const props = { name: 'Tom', age: null }
        const wrappedComponent = shallow(<User {...props} />);
        expect(wrappedComponent).toMatchSnapshot();
    }),

    it ('should render passed age', () => {
        const props = { name: 'Tom', age: 100 }
        const wrappedComponent = shallow(<User {...props} />);
        expect(wrappedComponent).toMatchSnapshot();
    })
    
})