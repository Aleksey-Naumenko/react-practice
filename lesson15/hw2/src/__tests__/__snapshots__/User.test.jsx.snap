// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<User /> should render passed age 1`] = `
<li
  className="user"
>
  <span
    className="user__name"
  >
    Tom
  </span>
  <span
    className="user__age"
  >
    100
  </span>
</li>
`;

exports[`<User /> should render passed name 1`] = `
<li
  className="user"
>
  <span
    className="user__name"
  >
    Tom
  </span>
  <span
    className="user__age"
  />
</li>
`;
